### PR TITLE
Layout

### PR DESCRIPTION
**Pull request by dtclinch**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by amcdnl** [comment]
@dtclinch - This looks interesting - happy to accept a pr

File: src/utils/useDrag.ts [hunk 1]
```diff
@@ -63,10 +63,17 @@ export const useDrag = ({
       },
       onDrag: ({ event }) => {
         // Compute normalized mouse coordinates (screen space)
+        const scrollX = window.scrollX || window.pageXOffset;
```
Comment (amcdnl) [inline src/utils/useDrag.ts:66]: @ghsteff - 👀

File: src/themes/theme.ts [hunk 2]
```diff
@@ -11,10 +11,16 @@ export interface Theme {
     opacity: number;
     selectedOpacity: number;
     inactiveOpacity: number;
+    showRing?: boolean;
```
Comment (amcdnl) [inline src/themes/theme.ts:14]: If you pass null to the node for the ring, it will hide it.

File: src/symbols/edges/Edges.tsx [hunk 3]
```diff
@@ -227,7 +227,7 @@ export const Edges: FC<EdgesProps> = ({
         <a.meshBasicMaterial
           attach="material-0"
           color={theme.edge.fill}
-          depthTest={false}
+          depthTest={true}
```
Comment (amcdnl) [inline src/symbols/edges/Edges.tsx:230]: Curious why this changed

File: src/symbols/Label.tsx [hunk 4]
```diff
@@ -1,7 +1,49 @@
 import React, { FC, useMemo } from 'react';
-import { Billboard, Text } from 'glodrei';
+import { Billboard, RoundedBox, Text } from 'glodrei';
 import { Color, ColorRepresentation, Euler } from 'three';
 import ellipsize from 'ellipsize';
+import { a } from '@react-spring/three';
+
+const calculateTextSize = (
```
Comment (amcdnl) [inline src/symbols/Label.tsx:7]: We have something similar to this in reaviz - https://github.com/reaviz/reaviz/blob/master/src/common/utils/size.ts#L8

**Comment by dtclinch** [comment]
@amcdnl My apologies, it was raised by mistake, currently a lot of things are under experiments and testing phase, we are trying to add features like Zoomcharts network graphs.

**Pull request closed without merging**